plugins/autowarmth.koplugin/main.lua

```diff
@@ -0,0 +1,780 @@
+--[[--
+@module koplugin.autowarmth
+
+Plugin for setting screen warmth based on the sun position and/or a time schedule
+]]
+
+local Device = require("device")
+
+local ConfirmBox = require("ui/widget/confirmbox")
+local DoubleSpinWidget = require("/ui/widget/doublespinwidget")
+local DeviceListener = require("device/devicelistener")
+local Dispatcher = require("dispatcher")
+local FFIUtil = require("ffi/util")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local Font = require("ui/font")
+local SpinWidget = require("ui/widget/spinwidget")
+local SunTime = require("suntime")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local _ = require("gettext")
+local T = FFIUtil.template
+local Screen = require("device").screen
+local util = require("util")
+
+local activate_sun = 1
+local activate_schedule = 2
+local activate_closer_noon = 3
+local activate_closer_midnight =4
+
+local midnight_index = 11
+
+local device_max_warmth = Device:hasNaturalLight() and Device.powerd.fl_warmth_max or 100
+local device_warmth_fit_scale = device_max_warmth / 100
+
+local function frac(x)
+    return x - math.floor(x)
+end
+
+local AutoWarmth = WidgetContainer:new{
+    name = "autowarmth",
+    easy_mode = G_reader_settings:isTrue("autowarmth_easy_mode") or false,
+    activate = G_reader_settings:readSetting("autowarmth_activate") or 0,
+    location = G_reader_settings:readSetting("autowarmth_location") or "Geysir",
+    latitude = G_reader_settings:readSetting("autowarmth_latitude") or 64.31, --great Geysir in Iceland
+    longitude = G_reader_settings:readSetting("autowarmth_longitude") or -20.30,
+    timezone = G_reader_settings:readSetting("autowarmth_timezone") or 0,
+    scheduler_times = G_reader_settings:readSetting("autowarmth_scheduler_times") or
+        {0.0, 5.5, 6.0, 6.5, 7.0, 13.0, 21.5, 22.0, 22.5, 23.0, 24.0},
+    warmth =   G_reader_settings:readSetting("autowarmth_warmth")
+        or { 90, 90, 80, 60, 20, 20, 20, 60, 80, 90, 90},
+    sched_times = {},
+    sched_funcs = {}, -- necessary for unschedule, function, warmth
+    sched_index = 1,
+}
+
+-- get timezone offset in hours (including dst)
+function AutoWarmth:getTimezoneOffset()
+    local utcdate   = os.date("!*t")
+    local localdate = os.date("*t")
+    return os.difftime(os.time(localdate), os.time(utcdate))/3600
+end
+
+function AutoWarmth:init()
+    self:onDispatcherRegisterActions()
+    self.ui.menu:registerToMainMenu(self)
+
+    -- schedule recalculation shortly afer midnight
+    self:scheduleMidnightUpdate()
+end
+
+function AutoWarmth:onDispatcherRegisterActions()
+    Dispatcher:registerAction("show_ephemeris",
+        {category="none", event="ShowEphemeris", title=_("Show ephemeris"), device=true})
+end
+
+function AutoWarmth:onShowEphemeris()
+    self:showTimesInfo(_("Infos about the sun in"), true, activate_sun, false)
```

Comment:
I don't see a context for this on the screenshots?
```suggestion
    self:showTimesInfo(_("Information about the sun in"), true, activate_sun, false)
```